indigo/indigo/src/main/scala/indigo/shared/FrameContext.scala

```diff
@@ -29,10 +30,12 @@ final class FrameContext[StartUpData](
     val dice: Dice,
     val inputState: InputState,
     val boundaryLocator: BoundaryLocator,
-    _startUpData: => StartUpData
+    _startUpData: => StartUpData,
+    _renderer: => Renderer
```

Comment:
My first reaction of "WOW that's scary 😱", after taking a moment, I'm not sure it's as scary as it looks, but this does still feel like a sledge hammer way to provide access to the screen capture methods.

An alternative could be:

```scala
final class FrameContext[StartUpData](
  ...
  _startUpData: => StartUpData,
  _captureScreen: Batch[ScreenCaptureConfig] => Batch[Either[String, AssetType.Image]]
```

And then you expose the two variants here on the framecontext an remove one of them from Renderer, i.e.

`Renderer` only has:
```scala
  def captureScreen(captureConfig: Batch[ScreenCaptureConfig]): Batch[Either[String, AssetType.Image]]
```

And `FrameContext` has both:

```scala
  def captureScreen(captureConfig: Batch[ScreenCaptureConfig]): Batch[Either[String, AssetType.Image]]

  def captureScreen(captureConfig: ScreenCaptureConfig): Either[String, AssetType.Image] =
    captureScreen(Batch(captureConfig)).headOption match {
      case Some(v) => v
      case None    => Left("Could not capture image")
    }
```

Reply:
Yes you're right, that's a better route here - I've updated it